qadence/blocks/time_block.py

```diff
@@ -8,73 +9,107 @@
 
 
 class TDGenerator:
-    def __init__(self, schedule: list[dict]) -> None:
+    def __init__(self, schedule: dict[int, dict]) -> None:
         self._schedule = schedule
 
     @property
     def duration(self) -> int:
         return self._schedule[-1]["t_end"]  # type: ignore [no-any-return]
 
     @property
-    def schedule(self) -> list[dict]:
+    def schedule(self) -> dict[int, dict]:
         return self._schedule
 
     @classmethod
     def from_block(cls, duration: int, block: AbstractBlock) -> TDGenerator:
-        schedule = [
-            {
+        schedule = {
+            duration: {
                 "t_start": 0,
                 "t_end": duration,
-                "generators": [block],
-                "fn": None,
+                "generator": block,
                 "qubit_support": block.qubit_support,
             }
-        ]
+        }
         return cls(schedule)
 
+    def _create_new_schedule(self, op: str, other: TDGenerator) -> dict[int, dict]:
+        if op in ["add", "kron"]:
```

Comment:
Shouldn't that be in the operator overloads ?

Reply:
There are overloads - `__mul__`, `__add__` and `__matmul__` are implemented just below. I just wanted to avoid duplication of code in case of `__matmul__` and `__add__` since the only difference is in one line.